### PR TITLE
Orchestration streaming

### PR DESCRIPTION
**Pull request by CharlesDuboisSAP**
## Context

AI/ai-sdk-java-backlog#71.

As an orchestration user I want to stream responses, similarly to streaming when using OpenAI

### Feature scope:
 
- [x] Added `streamChatCompletionDeltas()` and `streamChatCompletion()` to the `OrchestrationClient`
- [ ] Reduce code duplication to be done in AI/ai-sdk-java-backlog#111

## Definition of Done

- [x] Functionality scope stated & covered
- [x] Tests cover the scope above
- [x] Error handling created / updated & covered by the tests above
- [ ] ~Aligned changes with the JavaScript SDK~
- [x] Documentation updated
- [x] Release notes updated

### CODE REVIEW

File: orchestration/src/main/java/com/sap/ai/sdk/orchestration/OrchestrationClient.java [hunk 1]
```diff
@@ -111,6 +113,32 @@ public OrchestrationChatResponse chatCompletion(
...
+  /**
+   * Generate a completion for the given prompt.
+   *
+   * @param prompt a text message.
+   * @return A stream of message deltas
+   * @throws OrchestrationClientException if the request fails or if the finish reason is
+   *     content_filter
+   */
+  @Nonnull
+  public Stream<String> streamChatCompletion(
```
Comment (newtork) [inline orchestration/src/main/java/com/sap/ai/sdk/orchestration/OrchestrationClient.java:125]: (Comment)

New method names match the existing `OpenAiClient` counter-part in `openai` module.

File: orchestration/src/main/java/com/sap/ai/sdk/orchestration/OrchestrationStreamingHandler.java [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+  @SuppressWarnings("PMD.CloseResource") // Stream is closed automatically when consumed
+  @Nonnull
+  Stream<D> handleResponse(@Nonnull final ClassicHttpResponse response)
+      throws OrchestrationClientException {
+    if (response.getCode() >= 300) {
+      buildExceptionAndThrow(response);
+    }
+    return IterableStreamConverter.lines(response.getEntity())
+        // half of the lines are empty newlines, the last line is "data: [DONE]"
+        .peek(line -> log.info("Handler: {}", line))
```
Comment (newtork) [inline orchestration/src/main/java/com/sap/ai/sdk/orchestration/OrchestrationStreamingHandler.java:33]: (Comment)

I'm surprised we have so many `info` log statements. I feel like we could define a coding guideline when to use what log level.

(no action required.)

File: docs/guides/ORCHESTRATION_CHAT_COMPLETION.md [hunk 3]
```diff
@@ -186,6 +186,33 @@ var result =
...
+            System.out.print(deltaString);
+            System.out.flush();
+        });
+}
+```
+
+#### Spring Boot example
+
+Please find [an example in our Spring Boot application](../../sample-code/spring-app/src/main/java/com/sap/ai/sdk/app/controllers/OrchestrationController.java). It shows the usage of Spring
+Boot's `ResponseBodyEmitter` to stream the chat completion delta messages to the frontend in real-time.
```
Comment (newtork) [inline docs/guides/ORCHESTRATION_CHAT_COMPLETION.md:213]: (Minor)

Please add line-break after sentence.

File: orchestration/src/main/java/com/sap/ai/sdk/orchestration/OrchestrationClient.java [hunk 4]
```diff
@@ -211,4 +239,53 @@ CompletionPostResponse executeRequest(@Nonnull final String request) {
...
+      request.setEntity(new StringEntity(json, ContentType.APPLICATION_JSON));
+    } catch (final JsonProcessingException e) {
+      throw new OrchestrationClientException("Failed to serialize request parameters", e);
+    }
+  }
+
+  @Nonnull
+  private <D extends StreamedDelta> Stream<D> streamRequest(
+      final BasicClassicHttpRequest request, @Nonnull final Class<D> deltaType) {
+    val postRequest = new HttpPost("/completion");
```
Comment (newtork) [inline orchestration/src/main/java/com/sap/ai/sdk/orchestration/OrchestrationClient.java:280]: (Major)

Variable not used

File: docs/release-notes/release_notes.md [hunk 5]
```diff
@@ -12,7 +12,7 @@
 
 ### ✨ New Functionality
 
--
+- Added `streamChatCompletion()` and `streamChatCompletionDeltas()` to the `OrchestrationClient`
```
Comment (newtork) [inline docs/release-notes/release_notes.md:15]:
```suggestion
- Added `streamChatCompletion()` and `streamChatCompletionDeltas()` to the `OrchestrationClient`.
```

File: orchestration/src/test/java/com/sap/ai/sdk/orchestration/OrchestrationUnitTest.java [hunk 6]
```diff
@@ -273,8 +297,20 @@ void maskingPseudonymization() throws IOException {
...
-  @Test
-  void testErrorHandling() {
+  private static Runnable[] errorHandlingCalls() {
+    return new Runnable[] {
+      () -> client.chatCompletion(new OrchestrationPrompt(""), config),
+      () ->
+          client
+              .streamChatCompletion(new OrchestrationPrompt(""), config)
+              // the stream needs to be consumed to parse the response
+              .forEach(System.out::println)
```
Comment (newtork) [inline orchestration/src/test/java/com/sap/ai/sdk/orchestration/OrchestrationUnitTest.java:307]: (Question)

What's the difference between the two calls?  
Looks like it doesn't affect assertions(?)

Comment (CharlesDuboisSAP) [reply orchestration/src/test/java/com/sap/ai/sdk/orchestration/OrchestrationUnitTest.java:307]: `chatCompletion` <> `streamChatCompletion`

**Review by newtork — APPROVED** [review]

**Pull request merged**